Measurements and categories update (#60)
spiceai/docs: spiceaidocs/content/en/reference/pod/_index.md

    name: btcusd
```

### `dataspaces[*].measurements`

> The measurements to create in the dataspace. If these measurements are not populated by either a data connector or through the API, they will use the default value of `0.0`. Each field must be a `float`.

Reviewer: This is not 100% accurate right? They will be initialized to a default value of 0.0 then filled or not?

Author: That sentence was confusing so I removed it. I wanted to highlight that unless data is provided, it will default to zero.

Author: But I think that is intuitive enough to not have to call out explicitly

**Example**

```yaml
measurements:
  - name: balance
    initializer: 10.0
```

### `dataspaces[*].measurements[*].selector`

> Used to select which field in the data source to map to this measurement name. Defaults to `name` if not provided.

Reviewer: ```suggestion
Used to select which field in the data source to map to this measurement name. Defaults to the measurement name if not provided.
```

**Example**

In the source data:

```csv

> time,bal

Reviewer: I would make this `my_balance` or similar as `bal` might not be 100% clear.

100,10
200,20
```

This would "select" the `bal` column and place it into `balance` to be used by Spice.ai

```yaml
measurements:
  - name: balance
    selector: bal
```

### `dataspaces[*].measurements[*].fill`

Used to specify how to treat missing data. Possible values are `previous` or `none`. The default value is `previous`.

`previous` will take the last data point and "fill" it in. `none` will fill in the data with 0.

**Example**

In the source data:

```csv
time,bal
100,10
125,
150,15
175,
200,20
```

The following manifest:

```yaml
measurements:
  - name: bal
    fill: previous
```

would produce this data to the AI Engine:

```csv
time,bal
100,10
125,10
150,15
175,15
200,20
```

### `dataspaces[*].categories`

> Process categorical data in your data source. Categorical data is any data that can come from a finite number of values.

Reviewer: ```suggestion
Process categorical data. Categorical data is a group or collection of discrete strings.
```

> Some examples of categorical data include: colors, ratings, brands, programming languages, age group, hair color, grades, etc.

Reviewer: ```suggestion
Some examples of categorical data include: colors, star ratings, brands, programming languages, age group, hair color, grades, etc.
```

> To specify categorical data in your data source, use the `categories` node and specify the `values` that can appear in the category `name`.

Reviewer: ```suggestion
Define categorical collections in the Spicepod manifest by the categories` node. Specify each category with a `name` and a list of discrete `values.`
```

**Example**

```yaml
dataspaces:
  - from: rainbow
    name: attributes
    categories:
      - name: color
        values:
          - red
          - orange
          - yellow
          - green
          - blue
          - indigo
          - violet
```

### `dataspaces[*].categories[*].name`

> The `name` of the category. This will be the field that appears in the data source.

Reviewer: ```suggestion
Both the category name and the dataspace observation field name.
```

### `dataspaces[*].categories[*].selector`

> Used to select which field in the data source to map to this measurement name. Defaults to `name` if not provided.

Reviewer: ```suggestion
Selects the source data field mapped to the data space field. Defaults to `name` if not provided.
```

**Example**

In the source data:

```csv
time,color_rainbow
100,red
200,blue
```

> This would "select" the `color_rainbow` column and place it into `color` to be used by Spice.ai

Reviewer: ```suggestion
The data field `color_rainbow` is selected as the mapped source data for the data space observation field `color.`
```

```yaml
categories:
  - name: color
    selector: color_rainbow
```

### `dataspaces[*].categories[*].values`

> The set of possible categorical values that will be accepted by Spice.ai for the category `name`.

Reviewer: ```suggestion
Specifies the list of discrete category `name` values.
```

> Any values appearing in the data source that do not appear in this list will be ignored.

Reviewer: ```suggestion
Unspecified values in source data are ignored and are not included in the dataspace observations.
```

### `dataspaces[*].tags`

> A list of tags. Each item is a string value. If this value appears in the data source, it will be sent to the AI Engine in the observation.

Reviewer: ```suggestion
Specifies the list of tags. Each item is a string value. Unspecified tags in source data are ignored and are not included in the dataspace observations.
```

**Example**

```yaml
dataspaces:
  - from: coinbase
    name: btcusd
    tags:
      - buy
      - sell
```

### `dataspaces[*].data`

> Used to specify the external data source (configured with `connector` and `processor`) to use to populate the dataspace.

Reviewer: ```suggestion
Defines the data `connector` and data `processor` for the dataspace.
```
